packages/flutter_tools/lib/src/commands/drive.dart

```diff
@@ -136,11 +137,23 @@ class DriveCommand extends RunCommandBase {
   bool get shouldBuild => boolArg('build');
 
   bool get verboseSystemLogs => boolArg('verbose-system-logs');
+  String get userIdentifier => stringArg(FlutterOptions.kDeviceUser);
 
   /// Subscription to log messages printed on the device or simulator.
   // ignore: cancel_subscriptions
   StreamSubscription<String> _deviceLogSubscription;
 
+  @override
+  Future<void> validateCommand() async {
+    if (userIdentifier != null) {
```

Comment:
Will this break -d all with a user identifier?

Reply:
Before this change `flutter drive -d all` runs on the first one it finds.  I can't find an issue complaining about that.
```
$ flutter drive -d all -t test_driver/route_test.dart
Found multiple connected devices:
Pixel 4           • 98281FFAZ005T6                       • android-arm64 •
Android 10 (API 29)
iPhone 11 Pro Max • 16AF0D86-8F62-410C-8CC2-8F70C49835AF • ios           •
com.apple.CoreSimulator.SimRuntime.iOS-13-5 (simulator)
Using device Pixel 4.
...
```
This `validateCommand` Android check is against the one device that's about to be used in `runCommand` [`_device = await findTargetDevice();`](https://github.com/flutter/flutter/blob/1e4d9f8502b7911f920c11d289b807c0ec7afd1b/packages/flutter_tools/lib/src/commands/drive.dart#L151).